### PR TITLE
feat(ts/analyzer): Update instance of constructur ret_ty is any

### PR DESCRIPTION
**Pull request by sunrabbit123**
**Description:**

```ts
interface FConstructor {
  new (): any;
}
interface F {
  foo: string;
  bar: number;
}
declare var F: FConstructor;

var obj11: F | string;
if (obj11 instanceof F) {
  // can't type narrowing, construct signature returns any.
  obj11; // F | string Because F constructor is any
  obj11.foo; // Error NoSuchVar
}
```

But, increase extra_error in DestructureVariables

So we need to solve the #552 issue first.

### CODE REVIEW

File: crates/stc_ts_type_checker/tests/conformance/controlFlow/dependentDestructuredVariables.stats.rust-debug [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Stats {
     required_error: 2,
     matched_error: 0,
-    extra_error: 23,
+    extra_error: 35,
```
Comment (kdy1) [inline crates/stc_ts_type_checker/tests/conformance/controlFlow/dependentDestructuredVariables.stats.rust-debug:4]: I think this is fine (https://github.com/dudykr/stc/issues/552)

**Review by kdy1 — COMMENTED** [review]
Can you rebase?

File: crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs [hunk 2]
```diff
@@ -1269,6 +1271,7 @@ impl Analyzer<'_, '_> {
                 .map(|orig_ty| self.narrow_with_instanceof(span, ty.clone(), orig_ty))
                 .collect::<Result<Vec<_>, _>>()?;
 
+            dbg!(&new_types, &orig.types, &ty);
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:1274]: Please remove this

File: crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs [hunk 3]
```diff
@@ -538,6 +538,19 @@ pub(crate) struct AccessPropertyOpts {
...
+    //   foo: string;
+    //   bar: number;
+    // }
+
+    //  var obj11: F | string;
+    //
+    //  obj11.foo; // Error TS2339
+    //
+    // ```
+    pub disallow_property_is_any: bool,
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs:553]: Not sure about this name

File: crates/stc_ts_type_checker/tests/conformance/controlFlow/dependentDestructuredVariables.stats.rust-debug (hunk 1 above)
Comment (sunrabbit123) [reply crates/stc_ts_type_checker/tests/conformance/controlFlow/dependentDestructuredVariables.stats.rust-debug:4]: But, increase extra_error in DestructureVariables

So we need to solve the https://github.com/dudykr/stc/issues/552 issue first.

File: crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs (hunk 3 above)
Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs:553]: Then, do you have a name you recommend?

File: crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs (hunk 2 above)
Comment (sunrabbit123) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:1274]:
```suggestion
```

File: crates/stc_ts_type_checker/tests/conformance/controlFlow/dependentDestructuredVariables.stats.rust-debug (hunk 1 above)
Comment (kdy1) [reply crates/stc_ts_type_checker/tests/conformance/controlFlow/dependentDestructuredVariables.stats.rust-debug:4]: But I decided to focus on making inference correct, not the numbers

File: crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs (hunk 3 above)
Comment (kdy1) [reply crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs]: A property for such operation already exists, so I used it

**Review by kdy1 — APPROVED** [review]
Thanks!

File: crates/stc_ts_type_checker/tests/conformance/controlFlow/dependentDestructuredVariables.stats.rust-debug (hunk 1 above)
Comment (sunrabbit123) [reply crates/stc_ts_type_checker/tests/conformance/controlFlow/dependentDestructuredVariables.stats.rust-debug:4]: Yes, then we have to decide on the option name now.

**Comment by sunrabbit123** [comment]
test is dead @kdy1, please rerun this

**Pull request merged**